GitUI/GitUICommands.cs

```diff
@@ -2174,13 +2174,27 @@ private void RunMergeToolOrConflictCommand(Dictionary<string, string> arguments)
             return arguments;
         }
 
+        internal static Func<string, bool> GetFindFilePredicate(string name, string workingDir)
```

Comment:
Statics are evil because they make dependency injection and behaviour replacement impossible.
Not to mention this class has more than 2,000 lines of code and adding more to it is just wrong.

Please create a new SOLID class something like:

```
public interface IFindFilePredicateProvider
{
    Get(string name, string workingDir);
}

public sealed class FindFilePredicateProvider
{
    public Get(string name, string workingDir) { ... }
}
```

Then have it tested.